### ISSUE TITLE
Library unusable with react native 0.74.2

### ISSUE THREAD

**Issue by RustyRishii** [react-native-clipboard/clipboard#250]
## Environment

info Fetching system and libraries information...
System:
  OS: Windows 10 10.0.19045
  CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Memory: 1.26 GB / 7.85 GB
Binaries:
  Node:
    version: 20.11.0
    path: D:\Softwares\Software Setups\nodejs\node.EXE
  Yarn:
    version: 1.22.22
    path: ~\AppData\Roaming\npm\yarn.CMD
  npm:
    version: 10.2.4
    path: D:\Softwares\Software Setups\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK:
    API Levels:
      - "28"
      - "29"
      - "30"
      - "31"
      - "32"
      - "33"
      - "34"
    Build Tools:
      - 28.0.3
      - 29.0.2
      - 30.0.2
      - 30.0.3
      - 32.0.0
      - 33.0.0
      - 33.0.1
      - 34.0.0
      - 35.0.0
      - 35.0.0
    System Images:
      - android-30 | Google APIs Intel x86 Atom
      - android-30 | Google APIs ATD Intel x86 Atom
      - android-31 | Intel x86_64 Atom
      - android-33 | Google APIs ARM 64 v8a
      - android-33 | Google APIs Intel x86_64 Atom
      - android-34 | Google APIs ARM 64 v8a
      - android-34 | Google APIs Intel x86_64 Atom
    Android NDK: Not Found
  Windows SDK:
    AllowAllTrustedApps: Disabled
    AllowDevelopmentWithoutDevLicense: Enabled
    Versions:
      - 10.0.16299.0
      - 10.0.19041.0
IDEs:
  Android Studio: Not Found
  Visual Studio:
    - 16.11.32413.511 (Visual Studio Community 2019)
Languages:
  Java: 17.0.10
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.2
    wanted: 0.74.2
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

## Platforms
 Android

## Versions

- Android: 12
- iOS:
- react-native-netinfo:
- react-native: 0.74.2
- react: 

## Description

I created a react native project with the expo command npx create-expo-app@latest and chose the typescript blank template. I tried to install this library in the project and it outputs dependency conflicts. 

## Reproducible Demo

I installed this library with the command `npm i @react-native-clipboard/clipboard` 
This was the output:- 
![image](https://github.com/react-native-clipboard/clipboard/assets/63153873/11678633-3d69-4d8a-97b0-f215f0c05db9)

**Comment by IAmRahul123** [comment]
@RustyRishii , i am also facing the same issue, did you resolve this issue

**Comment by RustyRishii** [comment]
> @RustyRishii , i am also facing the same issue  in react native version 0.72.7, did you resolve this issue
> 

Nope, i created a new project of version 0.73 as I had to get something done immediately.

**Comment by arochedy** [comment]
we just need to update package.json of this library ? I can create a PR if it's just this

**Comment by foneazmi** [comment]
`npm i @react-native-clipboard/clipboard --legacy-peer-deps`

maybe can try this command @RustyRishii

**Comment by IAmRahul123** [comment]
If you want to use it, you can just download the previous version, it will work,

@foneazmi I already tried this command, with this you can install but can not use it, it will give you error.

**Comment by foneazmi** [comment]
@IAmRahul123 

maybe you can just use yarn
test install it using yarn 4.2.1 no issue..


RN 0.74.3
![image](https://github.com/react-native-clipboard/clipboard/assets/26783512/aa9bce01-3bc0-462d-9918-ec4f3123d02b)

**Comment by elenitaex5** [comment]
I'm facing this issue to trying to upgrade to Expo SKD 51 which is using RN 0.74.3

**Comment by xuelink** [comment]
```bash
xue:langx (main) $ npm install --save @react-native-clipboard/clipboard
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: langx-react-native@1.0.0
npm ERR! Found: react-native@0.74.2
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.74.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react-native@"^0.61.5 || ^0.62.3 || ^0.63.2 || ^0.64.2 || ^0.65.0 || ^0.66.0 || ^0.67.0 || ^0.68.0 || ^0.69.0 || ^0.70.0 || ^0.71.0 || ^0.72.0 || ^0.73.0" from @react-native-clipboard/clipboard@1.14.1
npm ERR! node_modules/@react-native-clipboard/clipboard
npm ERR!   @react-native-clipboard/clipboard@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/xl/.npm/_logs/2024-07-18T19_31_50_941Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /Users/xl/.npm/_logs/2024-07-18T19_31_50_941Z-debug-0.log
```

**Comment by MrRefactor** [comment]
Quick workaround is to create `overrides` object inside `package.json` as follows:
`  
"overrides": {
    "react-native": "0.74.2",
  },
  `

**Comment by MuriloBarranco** [comment]
+1

**Comment by Kato-147** [comment]
i had the same problem and i fixed it by running the command npm install @react-native-clipboard/clipboard@1.13.0 --legacy-peer-deps

**Comment by xuelink** [comment]
> Quick workaround is to create `overrides` object inside `package.json` as follows:
> 
> `  
> 
> "overrides": {
> 
>     "react-native": "0.74.2",
> 
>   },
> 
>   `
> 
>  

it fixes temporarily but we need a PR here to fix it permanently

**Comment by RustyRishii** [comment]
@arochedy  You can just change the react native version from package.json. Dont create a PR.

**Comment by hubertwang** [comment]
Limit the override scope to @react-native-clipboard/clipboard only.
Temporarily solution before new version is out
``` javascript
  "overrides": {
    "@react-native-clipboard/clipboard": {
      "react-native": "YOUR_REACT_NATIVE_VERSION"
    }
  }
```

**Comment by usmandilmeer** [comment]
I have solve this issue by installing this version ^1.13.2 of "npm i @react-native-clipboard/clipboard" in my react- native project version "0.74.5"

**Comment by kevinbzrra** [comment]
Installing @react-native-clipboard/clipboard version ^1.13.2 solves to me
using react native 0.74.3

**Comment by brifiction** [comment]
Thanks @usmandilmeer and @kevinbzrra! I too attempted the same temporary fix like you did, for a `react-native` = `0.74.5` repo.

Installing `@react-native-clipboard/clipboard` to `1.13.2` **temporarily** did solve the `yarn` installation & the "post-resolution validation" below:

```bash
➤ YN0060: │ react-native is listed by your project with version 0.74.5 (pd037b), which doesn't satisfy what @react-native-clipboard/clipboard and other dependencies request (~0.70.0 || ~0.71.0 || ~0.72.0 || ~0.73.0).
➤ YN0002: │ RNDemoApp@workspace:. doesn't provide react-native-macos (p55bfa), requested by @react-native-clipboard/clipboard.
➤ YN0002: │ RNDemoApp@workspace:. doesn't provide react-native-windows (p0de5f), requested by @react-native-clipboard/clipboard and other dependencies.
```

**Details**
Node: v22.5.1
Yarn: v4.4.0

**`package.json` file**
```json
  "dependencies": {
    "@react-native-clipboard/clipboard": "1.13.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.74.5"
  },
  "devDependencies": {
    "typescript": "5.0.4"
  },
  "packageManager": "yarn@4.4.0",
  "engines": {
    "node": ">=22.5.1",
    "yarn": ">=4.4.0"
  }
```

> Please note, additional testing required despite temporary fix. I am simply sharing my findings from suggested fix and `yarn install` output.
